src/engine/app-manager.ts

```diff
@@ -111,57 +112,54 @@ export abstract class AppManagerApi implements API<AppManager> {
 
   /** Activate a module or plugin */
   public activateOne(name: string) {
-    const entry = this.getEntity(name)
-    this.activateCallAndEvent(entry)
-    if (this.isPlugin(entry)) {
-      this.activateRequestAndNotification(entry)
+    const api = this.getEntity(name)
+    this.activateCallAndEvent(api)
+    if (this.isPlugin(api)) {
+      this.activateRequestAndNotification(api)
     }
-    if (entry.api.activate) {
-      entry.api.activate()
+    if (api.activate) {
+      api.activate()
     }
     this.setActive(name, true)
-    this.events.emit('activate', entry)
+    this.events.emit('activate', api)
   }
 
   /** Activation for Module and Plugin */
-  private activateCallAndEvent<T extends Api>({ profile, api }: Entry<T>) {
-    const events = profile.events || []
-    events.forEach(event => {
+  private activateCallAndEvent<T extends Api>(api: PluginApi<T>) {
+    const events = api.profile.events || []
+    events.forEach((event: keyof T['events']) => {
       if (!api.events) return
       api.events.on(event, (...payload: any[]) =>
-        this.broadcast(profile.name, event as string, payload),
+        this.broadcast(api.name, event as string, payload),
       )
     })
 
-    const methods = profile.methods || []
-    this.calls[profile.name] = {}
+    const methods = api.profile.methods || []
+    this.calls[api.name] = {}

```

Comment:
is this `api.profile.name` ?

Reply:
`api.name === api.profile.name`. I use that to make it easier to filter in a store. Both values are `readonly` so there is no risk that the value changes.